packages/react-components/src/components/TroubleshootingGuideErrorBar.tsx

```diff
@@ -0,0 +1,151 @@
+   *  window.open('https://contoso.com/permissions-troubleshooting', '_blank');
+   * ```
+   *
+   * @remarks
+   * if this is not supplied, the composite will not show a 'further troubleshooting' link.
+   */
+  onPermissionsTroubleshootingClick?: (permissionsState: {
+    camera: PermissionState;
+    microphone: PermissionState;
+  }) => void;
+  /**
+   * Optional callback to supply users with further troubleshooting steps for network issues
+   * experienced when connecting to a call.
+   *
+   * @example
+   * ```ts
+   * onNetworkingTroubleshootingClick?: () =>
+   *  window.open('https://contoso.com/network-troubleshooting', '_blank');
+   * ```
+   *
+   * @remarks
+   * if this is not supplied, the composite will not show a 'network troubleshooting' link.
+   */
+  onNetworkingTroubleshootingClick?: () => void;
+  /**
+   * strings related to trouble shooting guidance link and dismiss button
+   */
+  troubleshootingGuideStrings: _TroubleshootingGuideErrorBarStrings;
+}
+
+/**
+ * @internal
+ * A component to show device Permission/network connection related errors, contains link that leads to further trouble shooting guide
+ */
+export const _TroubleshootingGuideErrorBar = (props: _TroubleshootingGuideErrorBarProps): JSX.Element => {
+  const theme = useTheme();
+  // error bar strings
+  const localeStrings = useLocale().strings.errorBar;
+  const strings = props.strings ?? localeStrings;
+
+  const [dismissedErrors, setDismissedErrors] = useState<DismissedError[]>([]);
+
+  const {
+    onPermissionsTroubleshootingClick,
+    onNetworkingTroubleshootingClick,
+    permissionsState,
+    troubleshootingGuideStrings
+  } = props;
+
+  // dropDismissalsForInactiveErrors only returns a new object if `dismissedErrors` actually changes.
+  // Without this behaviour, this `useEffect` block would cause a render loop.
+  useEffect(
+    () => setDismissedErrors(dropDismissalsForInactiveErrors(props.activeErrorMessages, dismissedErrors)),
+    [props.activeErrorMessages, dismissedErrors]
+  );
+
+  const toShow = errorsToShow(props.activeErrorMessages, dismissedErrors);
+
+  return (
+    <Stack data-ui-id="error-bar-stack">
+      {toShow.map((error) => (
+        <MessageBar
+          {...props}
+          styles={messageBarStyle(theme, messageBarType(error.type))}
+          key={error.type}
+          messageBarType={messageBarType(error.type)}
+          messageBarIconProps={messageBarIconProps(error.type)}
+          actions={
+            <MessageBarButton
+              text={troubleshootingGuideStrings.buttonText}
+              styles={confirmButtonStyle(theme)}
+              onClick={() => {
+                setDismissedErrors(dismissError(dismissedErrors, error));
+              }}
+              ariaLabel={strings.dismissButtonAriaLabel}
+            />
+          }
+          isMultiline={false}
+        >
+          {onPermissionsTroubleshootingClick || onNetworkingTroubleshootingClick ? (
```

Comment:
what happens here if the `onPermissionsTroubleshootingClick` is set but not the `onNetworkingTroubleshootingClick` and the network error fires? won't the network error bar send the user to the permissions help screen?

Reply:
the way I'm reading the logic there though is if there is a error set for the error bar and one of those functions are set you will have the error string of the incorrect error with the function.